File: app/src/main/java/vergecurrency/vergewallet/service/repository/db/TransactionRepositoryDAO.java
package vergecurrency.vergewallet.service.repository.db;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Property;

@Entity(nameInDb = "transaction")
public class TransactionRepositoryDAO {

    @Id(autoincrement = true)
    private long id;
    @Property(nameInDb = "txid")
    private String txid ;
    @Property(nameInDb = "action")
    private String action ;
    @Property(nameInDb = "amount")
    private long amount;
    @Property(nameInDb = "fees")
    private long fees;
    @Property(nameInDb = "time")
    private long time;
    @Property(nameInDb = "confirmations")
    private long confirmations;
    @Property(nameInDb = "blockheight")
    private long blockheight ;
    @Property(nameInDb = "fee_per_kb")
    private long feePerKb ;
    @Property(nameInDb = "saved_address")
    private String savedAddress ;
    @Property(nameInDb = "created_on")
    private long createdOn;
    @Property(nameInDb = "message")
    private String message ;
    @Property(nameInDb = "address_to")
    private String addressTo;
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/repository/db/TransactionRepositoryDAO.kt
package vergecurrency.vergewallet.service.repository.db

import org.greenrobot.greendao.annotation.Entity
import org.greenrobot.greendao.annotation.Id
import org.greenrobot.greendao.annotation.Property

@Entity(nameInDb = "transaction")
class TransactionRepositoryDAO {

    @Id(autoincrement = true)
    private val id: Long = 0
    @Property(nameInDb = "txid")
    private val txid: String? = null
    @Property(nameInDb = "action")
    private val action: String? = null
    @Property(nameInDb = "amount")
    private val amount: Long = 0
    @Property(nameInDb = "fees")
    private val fees: Long = 0
    @Property(nameInDb = "time")
    private val time: Long = 0
    @Property(nameInDb = "confirmations")
    private val confirmations: Long = 0
    @Property(nameInDb = "blockheight")
    private val blockheight: Long = 0
    @Property(nameInDb = "fee_per_kb")
    private val feePerKb: Long = 0
    @Property(nameInDb = "saved_address")
    private val savedAddress: String? = null
    @Property(nameInDb = "created_on")
    private val createdOn: Long = 0
    @Property(nameInDb = "message")
    private val message: String? = null
    @Property(nameInDb = "address_to")
    private val addressTo: String? = null
}
